### PR TITLE
Added header to reading list panel

### PR DESCRIPTION
**Pull request by simonhong**
fix https://github.com/brave/brave-browser/issues/32953

Reading list header includes its icon & title.

Screenshot 2023-09-18 at 11 02 22 AM

image

Resolves 

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

See the linked issue.

### CODE REVIEW

**Comment by simonhong** [comment]
`--leo-color-container-background` is set for header's background color.
However, we still use upstream's reading list webui. So both background colors are slightly different.
We can see it from above screen shot.
@rebron @minkyeongshin Should we go with our colors like above screenshot till reading list webui is updated?
Or should apply same bg color to header area?

File: browser/ui/views/side_panel/brave_read_later_side_panel_view.cc [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  // we can set `true` here.
+  SidePanelUtil::GetSidePanelContentProxy(this)->SetAvailable(false);
+  observation_.Observe(web_view);
+}
+
+void BraveReadLaterSidePanelView::OnViewVisibilityChanged(
+    views::View* observed_view,
+    views::View* starting_view) {
+  // Once it becomes available, stop observing becuase its availablity is not
+  // not changed from now on.
```
Comment (sangwoo108) [inline browser/ui/views/side_panel/brave_read_later_side_panel_view.cc:108]: nit: duplicated

```suggestion
  // Once it becomes available, stop observing becuase its availablity is not
  // changed from now on.
```

File: browser/ui/color/brave_color_id.h [hunk 2]
```diff
@@ -54,7 +54,10 @@
     E_CPONLY(kColorSidebarButtonPressed)                      \
     E_CPONLY(kColorSidebarItemBackgroundHovered)              \
     E_CPONLY(kColorSidebarItemDragIndicator)                  \
-    E_CPONLY(kColorSidebarSeparator)
+    E_CPONLY(kColorSidebarSeparator) \
+    E_CPONLY(kColorSidebarPanelHeaderSeparator) \
+    E_CPONLY(kColorSidebarPanelHeaderBackground) \
+    E_CPONLY(kColorSidebarPanelHeaderTitle)
```
Comment (sangwoo108) [inline browser/ui/color/brave_color_id.h:60]: nit: `\` vertical alignment

**Review by sangwoo108 — APPROVED** [review]
LGTM 👍

**Comment by simonhong** [comment]
@sangwoo108 Updated. Thanks!

**Review by mkarolin — APPROVED** [review]
`strings`++

**Pull request merged**